client/extensions/woocommerce/app/products/product-form-variations-row.js

```diff
@@ -1,111 +1,229 @@
-						unit="in"
-						dimensions={ variation.dimensions }
-						onChange={ setDimension }
-						noWrap
+	removeImage = () => {
+		const { editProductVariation, product, variation } = this.props;
+		this.setState( {
+			placeholder: null,
+			transientId: null,
+			isUploading: false,
+			src: null,
+			id: null,
+		} );
+		editProductVariation( product, variation, { image: {} } );
+	}
+
+	renderImage = () => {
+		const { src, placeholder, isUploading } = this.state;
+
+		let image;
+		if ( src && ! isUploading ) {
+			image = (
+				<figure>
+					<ImagePreloader
+						src={ src }
+						placeholder={ placeholder && ( <img src={ placeholder } /> ) || ( <span /> ) }
 					/>
-					<div className="products__product-weight-input">
-						<FormTextInputWithAffixes
-							name="weight"
-							type="number"
-							suffix="g"
-							value={ variation.weight || '' }
-							onChange={ setWeight }
-							size="4"
+				</figure>
+			);
+		} else if ( isUploading ) {
+			image = (
+				<figure>
+					<img src={ placeholder || ( <span /> ) } />
+					<Spinner />
+				</figure>
+			);
+		}
+
+		const classes = classNames( 'products__product-form-variation-image', {
+			preview: null === src,
+			uploader: ! image,
+		} );
+
+		const removeButton = image && (
+			<Button
+				compact
+				onClick={ this.removeImage }
+				className="products__product-form-variation-image-remove">
+					<Gridicon
+						icon="cross"
+						size={ 24 }
+						className="products__product-form-variation-image-remove-icon" />
+			</Button>
+		);
+
+		return (
+			<div className={ classes }>
+				<ProductImageUploader
+					compact
+					multiple={ false }
+					onSelect={ this.onSelect }
+					onUpload={ this.onUpload }
+					onError={ this.onError }
+				>{image}</ProductImageUploader>
+				{removeButton}
+			</div>
+		);
+	}
+
+	render() {
+		const { variation, manageStock, translate } = this.props;
+		return (
+			<tr className="products__product-form-variation-row">
+				<td className="products__product-id">
+					<div className="products__product-name-thumb">
+						{ this.renderImage() }
+						<span className="products__product-name products__variation-settings-link" onClick={ this.showDialog }>
+							{ formattedVariationName( variation ) }
+						</span>
+					</div>
+				</td>
+				<td>
+					<FormCurrencyInput noWrap
+						currencySymbolPrefix="$"
+						name="price"
+						value={ variation.regular_price || '' }
+						placeholder="0.00"
+						onChange={ this.setPrice }
+						size="4"
+					/>
+				</td>
+				<td>
+					<div className="products__product-dimensions-weight">
+						<FormDimensionsInput
+							className="products__product-dimensions-input"
+							unit="in"
+							dimensions={ variation.dimensions }
+							onChange={ this.setDimension }
 							noWrap
 						/>
+						<div className="products__product-weight-input">
+							<FormTextInputWithAffixes
+								name="weight"
+								type="number"
+								suffix="g"
```

Comment:
FWIW, even though it says `g` here, the destination site (octopus outlet) treated the value provided as pounds.